Fix screen recording on ARM64 by using architecture detection

geirolafs opened the pull request:
Replaces GPU detection with architecture detection to fix screen recording on ARM64 systems.

  - ARM64: Use wf-recorder (wl-screenrec not available)
  - x86_64: Prefer wl-screenrec for performance, fallback to wf-recorder
  - Aligns script logic with existing package installation strategy

  Fixes screen recording on Apple Silicon/ARM64 systems like MacBook Pro M1 Max running Asahi Linux.

nilszeilon committed:
add guard for arm architecture

nilszeilon committed:
change omarchy repo

nilszeilon committed:
install lazydocker instead of the bin (x86-64 arch only)

nilszeilon committed:
install guide for MacOS dual boot with omarchy

nilszeilon committed:
comment out 1password stuff

nilszeilon committed:
fix menu commands with vulkan setting

nilszeilon committed:
Autoselect BARE if we install with OMARCHY_ARM

nilszeilon committed:
fix typo

nilszeilon committed:
update docs on how to handle errors installing.

nilszeilon committed:
You only have to spec OMARCHY_ARM

nilszeilon committed:
add considerations to the installation guide

nilszeilon committed:
install pipewire for audio

nilszeilon committed:
add noconfirm flag to install of pipewire (conflict with jack) and add asahi-audio for speaker support

nilszeilon committed:
fix typo

nilszeilon committed:
fix install url

nilszeilon committed:
auto-detect ARM arch

nilszeilon committed:
add arm to state for exclusions

nilszeilon committed:
source 1password configs only on x86

nilszeilon committed:
migrate state for arm users

nilszeilon committed:
Merge remote-tracking branch 'upstream/master'

nilszeilon committed:
change back to the omarchy repo

nilszeilon committed:
update installation guide

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
remove comment

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
combine conditional

nilszeilon committed:
remove architecture guard

nilszeilon committed:
preflight setup script for asahi-alarm users

nilszeilon committed:
add prerequisites for Omarchy on asahi linux

nilszeilon committed:
add comment on use of correct shaders for aarch64

nilszeilon committed:
move arm check from guard

nilszeilon committed:
move check to asahi preflight

nilszeilon committed:
set the repo to boot correctly

nilszeilon committed:
symlink the aarch64 conf file

nilszeilon committed:
remove noconfirm, fails install

nilszeilon committed:
add bash

nilszeilon committed:
pass env

nilszeilon committed:
remove inf loop

nilszeilon committed:
append the mirror instead of replacing

nilszeilon committed:
pass full path

nilszeilon committed:
run install script again

nilszeilon committed:
reboot from new user

nilszeilon committed:
instructions to continue

nilszeilon committed:
install audio first to mitigate conflict

nilszeilon committed:
add noconfirm on pipewire things

nilszeilon committed:
complete automation of asahi setup of omarchy

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
override conf files with symlink

nilszeilon committed:
move asahi package to asahi pre-flight

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
use pacman here

nilszeilon committed:
remove mirror list and unused arm state

nilszeilon committed:
remove separate audio install file

nilszeilon committed:
add pipewire packages

nilszeilon committed:
remove ref to audio.sh

nilszeilon committed:
set boot repo

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
Merge remote-tracking branch 'upstream/dev' into add-aarch64-support

nilszeilon committed:
Merge branch 'add-aarch64-support'

nilszeilon committed:
Merge branch 'add-aarch64-support'

nilszeilon committed:
remove omarchy repo & add archlinuxarm.org as mirror

nilszeilon committed:
add back yay

nilszeilon committed:
add back yay

nilszeilon committed:
separate packages not found for aarch64

nilszeilon committed:
separate packages not found for aarch64

nilszeilon committed:
install pipewire-jack first

nilszeilon committed:
remove comment here

nilszeilon committed:
skip installing yay

nilszeilon committed:
move complicated packages

nilszeilon committed:
move localsend and yaru

nilszeilon committed:
fix eof issue

nilszeilon committed:
readd chromium

nilszeilon committed:
readd tte

nilszeilon committed:
add more mirrors

nilszeilon committed:
install tte via git

nilszeilon committed:
update repos

nilszeilon committed:
switch back to walker-bin

nilszeilon committed:
add small guide in asahi.sh for mac users

jondkinney committed:
Use omarchy-chromium-bin (universal), sort packages by name

jondkinney committed:
Add asahi-alarm/widevine package to support DRM content in chromium

This is necessary for spotify to work in the browser since there isn't
an arm64 compatible desktop version of spotify at the momemtn.

nilszeilon committed:
Merge pull request #1 from jondkinney/feat/add-omarchy-chromium-bin-and-widevine

Feat/add omarchy chromium bin and widevine

jondkinney committed:
Add asahi-alarm/widevine package to support DRM content in chromium

This is necessary for spotify to work in the browser since there isn't
an arm64 compatible desktop version of spotify at the moment.

jondkinney committed:
Add obsidian appimage which will pull arm64 version from AUR

nilszeilon committed:
Merge pull request #2 from jondkinney/fix/add-omarchy-chromium-bin-widevine-obsidian

Fix/add omarchy chromium bin widevine obsidian

jondkinney committed:
Remove asahi-alarm/widevine for x86

nilszeilon committed:
remove previous logic of overriding files, add patch hypr envs for aarch64

nilszeilon committed:
add pipewire packages to asahi

jondkinney committed:
Fix vanilla aarch64 install pkg list vs asahi-alarm install pkg list

jondkinney committed:
Fix pipewire-jack replacement by echoing "y" when prompted

https://www.reddit.com/r/archlinux/comments/1b3k635/which_is_better_jack2_or_pipewirejack/

jondkinney committed:
Fix walker

jondkinney committed:
Fix obsidian install, can't use AUR, need to download AppImage directly

jondkinney committed:
Skip limine-snapper.sh for ARM systems

jondkinney committed:
Update packages

jondkinney committed:
Support ARM mirrors

jondkinney committed:
ARM specific pacman.conf

jondkinney committed:
Guard symlinks in theme.sh by ensuring Yaru is avail

jondkinney committed:
Install specific walker version manually

jondkinney committed:
Put terminal text effects back

jondkinney committed:
Disable screensaver on virtual machines from hyperidle

jondkinney committed:
Update windowrulev2 for noscreenshare

jondkinney committed:
Update omarchy-version binary to work when changes are made to a local clone

jondkinney committed:
Install specific omarchy-chromium-bin for ARM

jondkinney committed:
Add pre-built walker

jondkinney committed:
Fix walker install

jondkinney committed:
Update obsidian icon

jondkinney committed:
Add 1password

jondkinney committed:
Add asdcontrol prebuilt arm binary and install script

jondkinney committed:
Add omarchy bin to path for ssh sessions

jondkinney committed:
Add xterm-256color for ssh sessions

jondkinney committed:
Handle asahi-alarm/widevine in a more integrated "omarchy" way

jondkinney committed:
Update install to continue automatically rather than exit 0, plus formatting

jondkinney committed:
Fix continuation. Exit is necessary, re-execute with existing env vars as created user

nilszeilon committed:
add arm/asahi installations back after merge

nilszeilon committed:
update docs on how to install

jondkinney committed:
Fix merge conflicts & formatting

jondkinney committed:
Restore basecamp/omarchy as OMARCHY_REPO

jondkinney committed:
Remove old packages install script

jondkinney committed:
Update how we handle Yaru for ARM

jondkinney committed:
Reorder pacman after OMARCHY_ARM export

jondkinney committed:
Fix terminal corruption

jondkinney committed:
test

jondkinney committed:
test2

jondkinney committed:
Move yay

jondkinney committed:
Force user selection / creation

jondkinney committed:
pink?

jondkinney committed:
more

jondkinney committed:
remove log pause check

jondkinney committed:
test suppress

doubt it

jondkinney committed:
validations

jondkinney committed:
echos

jondkinney committed:
Handle custom omarchy-lazyvim manually for ARM

jondkinney committed:
Fix pacman and yay flags

jondkinney committed:
Fix mkinitcpio hard fail from warnings

jondkinney committed:
Add limine for non-asahi arm systems

jondkinney committed:
fix limine arm install

jondkinney committed:
remove limine arm install

jondkinney committed:
Try to preserve env vars

jondkinney committed:
Update pacman post install to not clobber the arm config and mirrors

jondkinney committed:
remove conditional for yaru

jondkinney committed:
Skip login/limine-snapper.sh on ARM

jondkinney committed:
Limine boot support arm64

jondkinney committed:
add widevine back

jondkinney committed:
Fix 1password arm, not just arch

jondkinney committed:
Update repo order, asahi at the end

geirolafs committed:
Fix screen recording on ARM64 by using architecture detection
- Replace GPU detection (lspci grep nvidia) with architecture detection (uname -m)
- ARM64: Use wf-recorder (wl-screenrec not available)
- x86_64: Prefer wl-screenrec, fallback to wf-recorder
- Aligns script logic with existing package installation strategy

Fixes screen recording on Apple Silicon/ARM64 systems.

jondkinney commented on bin/omarchy-cmd-screenrecord:
Thanks for the contribution! I think it may be cleaner if we implemented it like this:

```bash
start_screenrecording() {
  filename="$OUTPUT_DIR/screenrecording-$(date +'%Y-%m-%d_%H-%M-%S').mp4"
  # Use wl-screenrec on x86_64 unless NVIDIA is present (compatibility issues)
  if [[ $(uname -m) == "x86_64" ]] && command -v wl-screenrec >/dev/null 2>&1 && ! lspci | grep -qi 'nvidia'; then
    wl-screenrec $AUDIO -f "$filename" --ffmpeg-encoder-options="-c:v libx264 -crf 23 -preset medium -movflags +faststart" "$@" &
  else
    wf-recorder $AUDIO -f "$filename" -c libx264 -p crf=23 -p preset=medium -p movflags=+faststart "$@" &
  fi
  toggle_screenrecording_indicator
}
```

What do you think?

geirolafs replied:
Agreed!

jondkinney committed:
Merge branch 'omarchy-dev' into amarchy-3-x

geirolafs committed:
Use Jon's improved logic for screen recorder selection

geirolafs committed:
Resolve merge conflict - use Jon's GPU detection approach

geirolafs committed:
Fix: Properly implement Jon's hybrid approach for ARM64 compatibility

  - Remove merge conflict markers that were accidentally left in file
  - Use architecture + tool availability + GPU detection as suggested
  - Only use wl-screenrec on x86_64 non-NVIDIA systems where it exists
  - Use wf-recorder for ARM64, NVIDIA, and fallback cases
  - This actually fixes screen recording on ARM64 systems
